source/type/CMakeLists.txt

```diff
@@ -27,38 +27,62 @@ SET(PROTO_SRCS ${CMAKE_CURRENT_BINARY_DIR}/chaos.pb.cc
+SET(MONITOR_KRAKEN_PROTO_PY
+    ${MONITOR_KRAKEN_DIR}/request_pb2.py
+    ${MONITOR_KRAKEN_DIR}/response_pb2.py
+    ${MONITOR_KRAKEN_DIR}/type_pb2.py
+)
+
+SET(JORMUNGANDR_TEST_DIR ${CMAKE_SOURCE_DIR}/jormungandr/tests)
+SET(JORMUNGANDR_TEST_PY
+    ${JORMUNGANDR_TEST_DIR}/chaos_pb2.py
+    ${JORMUNGANDR_TEST_DIR}/gtfs_realtime_pb2.py
+    ${JORMUNGANDR_TEST_DIR}/kirin_pb2.py
+)
+
 add_custom_command(OUTPUT ${PROTO_HDRS} ${PROTO_SRCS}
     COMMAND protoc ARGS ${PROTO_FILES} -I ${CMAKE_SOURCE_DIR}/navitia-proto -I ${CMAKE_SOURCE_DIR}/chaos-proto
     --cpp_out=${CMAKE_CURRENT_BINARY_DIR}
     WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/type" VERBATIM
     DEPENDS ${PROTO_FILES})
 
-add_custom_target(protobuf_files DEPENDS ${PROTO_HDRS} ${PROTO_SRCS})
-set_source_files_properties(${PROTO_HDRS} ${PROTO_SRCS} PROPERTIES GENERATED TRUE)
-
-
-add_library(pb_lib ${PROTO_SRCS} pb_converter.cpp)
-target_link_libraries(pb_lib vptranslator pthread ${PROTOBUF_LIBRARY})
-add_custom_command (TARGET pb_lib
-    PRE_BUILD
-    COMMAND protoc ARGS "--python_out=${CMAKE_SOURCE_DIR}/navitiacommon/navitiacommon" type.proto
-    response.proto request.proto task.proto stat.proto
-    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/navitia-proto" VERBATIM)
-
+add_custom_command (OUTPUT ${NAVITIACOMMON_PROTO_PY}
+    COMMAND protoc ARGS "--python_out=${NAVITIACOMMON_DIR}"
+    type.proto response.proto request.proto task.proto stat.proto
+    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/navitia-proto" VERBATIM
+    DEPENDS ${PROTO_FILES})
 
 #we don't want a dependancy to navitiacommon for monitor
-add_custom_command (TARGET pb_lib
-    PRE_BUILD
-    COMMAND protoc ARGS "--python_out=${CMAKE_SOURCE_DIR}/monitor/monitor_kraken" type.proto
-    response.proto request.proto
-    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/navitia-proto" VERBATIM)
+add_custom_command (OUTPUT ${MONITOR_KRAKEN_PROTO_PY}
+    COMMAND protoc ARGS "--python_out=${MONITOR_KRAKEN_DIR}"
+    type.proto response.proto request.proto
+    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/navitia-proto" VERBATIM
+    DEPENDS ${PROTO_FILES})
 
 #for integration tests, we need chaos protobuf
-add_custom_command (TARGET pb_lib
-    PRE_BUILD
-    COMMAND protoc ARGS "--python_out=${CMAKE_SOURCE_DIR}/jormungandr/tests" 
+add_custom_command (OUTPUT ${JORMUNGANDR_TEST_PY}
+    COMMAND protoc ARGS "--python_out=${JORMUNGANDR_TEST_DIR}"
     chaos.proto gtfs-realtime.proto kirin.proto
-    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/chaos-proto" VERBATIM)
+    WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/chaos-proto" VERBATIM
+    DEPENDS ${PROTO_FILES})
+
+add_custom_target(protobuf_files DEPENDS ${PROTO_HDRS} ${PROTO_SRCS}
+                  ${NAVITIACOMMON_PROTO_PY} ${MONITOR_KRAKEN_PROTO_PY} ${JORMUNGANDR_TEST_PY})
+set_source_files_properties(${PROTO_HDRS} ${PROTO_SRCS} PROPERTIES GENERATED TRUE)
+
+
+add_library(pb_lib ${PROTO_SRCS} pb_converter.cpp ${NAVITIACOMMON_PROTO_PY} ${MONITOR_KRAKEN_PROTO_PY} ${JORMUNGANDR_TEST_PY})
```

Comment:
So : added "lonely" to the `all` target, and still no dependency between `all` and `test` (like there is no dependency between `main_routing_test` and `jormungandr_integration_test`).
The advice is still to run `make && make test` instead of simply `make test`...